capstone/src/main/webapp/flags_script.js

```diff
@@ -143,16 +143,21 @@ function generateMap() {
   });
 }
  
-/* Populate the maps with flags. */
+/* Populate the maps with flags from the data in datastore. */
 async function getFlags() {
+  checkLogin(); //call function to hide login/logout buttons
+  deleteExpiredFlags(); //delete flags that are more 14 days old
```

Comment:
how often does this run, how many flags would it potentially load, is it only deleting expired flags for this user? might be okay to leave as is for now, but if you were building this for production, you would want a job separate from the web app dealing with deleting the expired flags

Reply:
That makes sense. For right now, the getFlags() method gets called once every time the website is loaded, not for every flag added. I will try to figure out how to have datastore delete entities serparately

Reply:
You could just pull flags, check that they were created in the past two weeks, and then have a separate regular batch process for deleting expired flags.